controllers/datadogagent/utils.go

```diff
@@ -973,12 +973,17 @@ func getVolumesForAgent(dda *datadoghqv1alpha1.DatadogAgent) []corev1.Volume {
 	}
 
 	if datadoghqv1alpha1.BoolValue(dda.Spec.Agent.Log.Enabled) {
+
+		var tempStoragePath = *dda.Spec.Agent.Log.TempStoragePath
+		if options.AgentHostStoragePath != "" {
+			tempStoragePath = filepath.Join(options.AgentHostStoragePath, "logs")
+		}
 		volumes = append(volumes, []corev1.Volume{
```

Comment:
I'm wondering if we can hide this behind a function if we go for a variadic options approach

```golang
package options

func GetHostStoragePath(options ...ReconcilerOptions, defaultPath string) string
```

Reply:
In any case we should add a helper to abstract it a bit.